python/orca/src/bigdl/orca/data/image/utils.py

```diff
@@ -167,9 +167,9 @@ def chunks(iterable: List[Dict[str, Union[str, int, float, "ndarray"]]],
 def pa_fs(path):
     import pyarrow as pa
     if path.startswith("hdfs"):  # hdfs://url:port/file_path
-        fs = pa.hdfs.connect()
-        path = path[len("hdfs://"):]
-        return path, fs
+        with pa.hdfs.connect() as fs:
```

Comment:
Need to add the methods `__enter__()` and `__exit__()` in the object methods if you want to use with statement. Otherwise there maybe `AttributeError: __enter__`